docs: consider ctrl clicking an anchor (#12290)
sveltejs/kit: documentation/docs/30-advanced/67-shallow-routing.md

	<a
		href="/photos/{thumbnail.id}"
		on:click={async (e) => {

> 			// bail if the screen is too narrow/small
> 			if (innerWidth < 640) return;
> 
> 			// bail if the link is opened in a new window
> 			if (e.shiftKey) return;
> 
> 			// bail if the link is opened in a new tab
> 			// macOS uses the metaKey, windows/linux use the ctrlKey
> 			if (e.metaKey || e.ctrlKey) return;
> 
> 			// add more bail checks if needed.
> 			// e.g. clicking with a mouse scroll wheel

Reviewer: while it is readable it also seems to be taking focus from the shallow routing portion. what about writing it like this?
```suggestion
			if (innerWidth < 640        // bail if the screen is too small
				|| e.shiftKey             // or the link is opened in a new window
				|| e.metaKey || e.ctrlKey // or a new tab (mac: metaKey, win/linux: ctrlKey)
				// should also consider clicking with a mouse scroll wheel 
			) return;
```

		href="/photos/{thumbnail.id}"
		on:click={async (e) => {

> 			// bail if opening a new tab, or we're on too small a screen

Reviewer: It might be nice to expand this with some details about ctrl/meta on various OS to explain why we need to check both

Author: Does <kbd>Windows</kbd> + clicking a link open it in a new tab?

Cannot reproduce it in Windows 11 + Chrome | Firefox.

Reviewer 2: It’s ctrl + click to open in a new tab on windows iirc

Reviewer 3: `Windows`+click (for me, on Firefox) on a regular link does nothing - it doesn't follow it at all. It looks like it _does_ trigger a `click` event, though, so to match that behavior, we should also make sure that we bail here and let the browser use the default handler, which is to do nothing.

_However_, there are also a number of other cases we should be handling here besides meta-click. Shift-click opens in a new window, and so we want to use the default handling there. Alt-clicking (for me) does nothing, so we also want the default handling there. I'm uncertain whether we need to check that this was a left-button click.

Author: I believe a scroll-wheel clicking on an anchor also opens it in a new tab as well. Should this be handled as well?

Reviewer: adding support for scroll-wheel clicking would be a nice touch

> 			// bail if opening a new tab, or we're on too small a screen

Reviewer: ```suggestion
			// bail if opening a new tab, or we're on too small a screen
			// macOS uses metaKey. windows and linux use ctrlKey
```

Author: Since Chrome uses the 'open link in new tab' context menu text, how about the following?

```diff
- for clicking to open a new tab,
+ for opening a link in a new tab,
+ to open the clicked link in a new tab,
``

Reviewer: I tweaked the suggestion a bit. what do you think about it now?

Author: That is intuitive. Can I add 'the' in-front of the key names?

> macOS uses the metaKey while windows and linux use the ctrlKey

Following citation is from Wikipedia.

> Similarly to the [Shift key](https://en.wikipedia.org/wiki/Shift_key), the Control key rarely performs any function when pressed by itself. The Control key is located on or near the bottom left side of most keyboards (in accordance with the international standard [ISO/IEC 9995-2](https://en.wikipedia.org/wiki/ISO/IEC_9995#ISO/IEC_9995-2)), with many featuring an additional one at the bottom right.

Reviewer: > Can I add 'the' in-front of the key names?

It does read nicely with 'the', but I was kind of trying to keep it on one line. We might have to break it into two lines if it gets too long, so I don't know if it'd be worth it or not

Author: I personally think splitting each checks is more readable.

```js
// bail if the user wants to open the link in a new tab
// macOS uses the metaKey, windows/linux use the ctrlKey
if (e.metaKey || e.ctrlKey) return;

// bail if the user wants to open the link in a new window
if (e.shiftKey) return;

// bail if the screen is too narrow/small
if (innerWidth < 640) return;
```

It might not be the most concise code, but there are more checks to add.

Will be adding this commit for now (also since `e.metaKey` has to be reverted)

> 			if (e.ctrlKey || innerWidth < 640) return;

Reviewer: why was `metaKey` removed?

Author: It was a mistake. Was planning to revert it after the discussion (Note: `metaKey` is used in macOS)
